Scheduler Resources demo: rewrite (#872)
DevExpress/devextreme-demos: JSDemos/Demos/Scheduler/Resources/description.md

### Define a Resource Kind Object in the [resources](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/') Property
The resource property is an array that stores resource kind objects. Each object should have at least the following fields:

- [dataSource](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/dataSource.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#dataSource')         

>     Maps a resource kind to resource instances. For details on how to provide data, see the [Data Binding](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Local_Array/) help topics.

Reviewer: To be honest, I don't get what the first sentence means. Doesn't the dataSource bind a resource to data?

- [fieldExpr](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/fieldExpr.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr')             

>     Specifies the name of an appointment data field that assigns the appointment to a specific category id.

Reviewer: The word category seems a bit confusing here. Maybe rephrase this sentence a bit to use the word 'resource' instead?

> To enable multiple category selection, set the object's [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property to **true**. 

Reviewer: The same question about category

> A resource (resource kind) adds an appointment field that allows users to assign the appointment to a single or multiple categories. The appointment rectangle will have the color based on the selected category. Scheduler can have multiple resources, but it derives the color scheme from the last resource kind object. You can use a color scheme of another object if you set its [useColorAsDefault](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#useColorAsDefault) property to **true**.

Reviewer: 1. Why add 'resource kind' in this sentence? 

2. >it derives the color scheme from the last resource kind object

Last object in the `resources[]` array?

To add a resource, do the following:

> ### Add resource instances

Reviewer: Let's use the title case here and below

> Resource instances are objects that define the available categories. Each resource instance object must contain the id, text, and color fields. Create an array of such objects. If your field names differ, specify them in the [valueExpr](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/valueExpr.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#valueExpr'), [colorExpr](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/colorExpr.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#colorExpr') and [displayExpr](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/displayExpr.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#displayExpr') properties.

Reviewer: It is not completely clear where users should specify these properties. Also please add the Oxford comma.

### Define a resource kind object in the [resources](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/') property
The resource property is an array that stores resource kind objects. Each object should have at least the following fields:

- [dataSource](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/dataSource.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#dataSource')         
    Maps a resource kind to resource instances. For details on how to provide data, see the [Data Binding](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Local_Array/) help topics.
- [fieldExpr](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/fieldExpr.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr')             
    Specifies the name of an appointment data field that assigns the appointment to a specific category id.

> Create one resource kind object. To enable multiple category selection, set the object's [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property to **true**. 

Reviewer: You've already described how to create such object. This sentence would be more appropriate in a tutorial, IMHO.

```suggestion
To enable multiple category selection, set the object's [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property to **true**. 
```

> By now, users should be able to select categories in the newly created appointment form field.

Reviewer: The same note about tutorial as above.

> ### Optionally, preset appointments to specific categories in code
> To associate a specific appointment with the resource, include a field specified in the **fieldExpr** property into the appointment's data object. This field should be equal to one of the appointment instances' id.

Reviewer: We can preset something only in code, right? I also don't think that the word optionally is a good choice for a heading. The last point - Ray doesn't like apostrophes after plurals.

```suggestion
### Preset appointments to specific categories
To associate a specific appointment with the resource, include a field specified in the **fieldExpr** property into the appointment's data object. This field's should be equal to id of one of the appointment instances.
```

> Users can categorize appointments by resources. To understand what resources are, consider the following example: in an educational center lectures are held in several rooms. In Scheduler terms, room is a resource kind, individual rooms are resource instances, and lectures are appointments assigned to the resource instances.

Reviewer 2: ```suggestion
Users can categorize appointments by resources. To understand what resources are, consider the following example: in an educational center lectures are held in several rooms. In Scheduler terms, room is a resource kind, individual rooms are resource instances, and lectures are appointments that use these resource instances.
```

> The Scheduler can have multiple resources. To add a resource, do the following:

Reviewer 2: The subheadings are self-explanatory.

```suggestion
```

> ### Define Resources

Reviewer 2: We agreed upon the terms we're going to use, now let's use them.

```suggestion
### Define Resource Kinds
```

1. **Add a resource object to the [resources](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/') array property.**        

> Each object should have at least the following fields:

Reviewer 2: > array property

So is this an array or property?

It's not a good idea to create a two-item list where the first item contains a nested list and the second item is optional.

```suggestion
Use the [resources](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/) array to define resource kinds. Each object in this array should contain at least the following fields:
```

- [dataSource](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/dataSource.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#dataSource')         
    Specifies available resource instances.          

>     Each resource instance object should contain the id, text, and color fields. If your field names differ, specify them in the global [valueExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#valueExpr), [colorExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#colorExpr), and [displayExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#displayExpr) properties. For details on how to provide data of resource instances, see the [Data Binding](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Local_Array/) help topics.

Reviewer 2: Crossed out unnecessary words. We need to be succinct, especially when giving explanations in lists.

```suggestion
- [dataSource](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#dataSource)         
    Resource instances of this resource kind. Each instance should contain the `id`, `text`, and `color` fields. If your field names differ, specify them in the [valueExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#valueExpr), [displayExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#displayExpr), and [colorExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#colorExpr) properties, respectively.

```

- [fieldExpr](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources/fieldExpr.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr')             
    Specifies the name of an appointment data field that sets a resource instance id.

1. **Preset appointments to specific resource instances.**           

> Include a field specified in the **fieldExpr** property into appointment data objects. The field should be equal to a resource instance id. If you skip this step, users should specify resource instances in the UI.

Reviewer 2: "Preset appointments to specific resource instances" is the main use case for fieldExpr. It should be described in the fieldExpr description.

This is also a good place to refer to the demo code.

```suggestion
- [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr)             
    A data field used to assign instances of this resource kind to appointments. Add this field to appointment objects and set its value to resource instance `id`s.

In this demo, the **resources** array contains three resource kinds: rooms, priorities, and owners. Their **fieldExpr** values are `roomId`, `priorityId`, and `ownerId`. Each appointment contains the same fields. Field values assign the appointments to different instances of these resource kinds.
```

### Use Color Scheme of a Custom Resouce Kind

> The Scheduler uses the color scheme of the last resource kind object in the **resources** array. If you need colors of another object, set its [useColorAsDefault](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#useColorAsDefault) property to **true**.

Reviewer 2: If you need to describe the default state of things, see if you can start with describing how to change it and mention the default values in the end. Note how I did this below.

Also, don't forget to demo.

```suggestion
### Color Appointments Based on Resource Kind

To use the color scheme of a specific resource kind, enable the kind's [useColorAsDefault](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#useColorAsDefault) property. Otherwise, appointments use the color scheme of the last resource kind declared in the **resources** array.

This demo enables you to change the **useColorAsDefault** property at runtime. Click the radio buttons under the Scheduler to switch between different color schemes.
```

> ### Enable Assignment of Multiple Resource Instances
> To enable assignment of multiple resource instances, set the resource object's [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property to **true**.

Reviewer 2: Multiple instances of a **single resource kind** - this is important.
We need to do a better job explaining what is what, not just "to enable this, do that".

```suggestion
### Assign Multiple Instances of a Resource Kind

Each resource kind object can contain the [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property. When it is **true**, users can assign multiple instances of this kind to a single appointment. In this demo, the Owner resource kind allows multiple instances.
```

> You can also group appointments by resources as specified in the following article: [Group Appointments by Resources](/Documentation/Guide/UI_Components/Scheduler/Resources/Group_Appointments_by_Resources/). 

Reviewer 2: We have a demo that's gonna be the main source of information on this scenario:

```suggestion
You can also group appointments by resources as shown in the following demo: [Group Orientation](/Demos/WidgetsGallery/Demo/Scheduler/GroupOrientation/). 
```

> Users can categorize appointments by resources. To understand what resources are, consider the following example: in an educational center lectures are held in several rooms. In Scheduler terms, room is a resource kind, individual rooms are resource instances, and lectures are appointments that use these resource instances.

Reviewer 3: ```suggestion
Users can categorize appointments by resources. The following example illustrates what resources are: in an educational center lectures are held in several rooms. In Scheduler terms, room is a resource kind, individual rooms are resource instances, and lectures are appointments that use these resource instances.
```

> ### Define Resource Kinds

Reviewer 4: Привет :) Я нигде не нашла такого термина Resource Kinds, везде они называются Resource Types (я просто гуглить пробовала). Может быть переименовать? Если же это термин, то я не возражаю)

> Use the [resources](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/) array to define resource kinds. Each object in this array should contain at least the following fields:

Reviewer 4: Как именно можно использовать этот массив? В него нужно добавлять объекты с указанными полями? Если так, то может быть так и сказать "Добавьте все необходимые типы ресурсов в такой-то массив?" Или как-то подобный образом.

Плюс, мне кажется, из второго предложения можно убрать at least (should вроде бы и так сожержит нужный смысл):
Each object in this array should contain the following fields:

Reviewer 3: Why `kinds` and not **types**?

Author: "Resource kind" is a term. It is introduced in the first paragraph of this text.

Reviewer 3: Ok

- [dataSource](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#dataSource)         
    Resource instances of this resource kind. Each instance should contain the `id`, `text`, and `color` fields. If your field names differ, specify them in the [valueExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#valueExpr), [displayExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#displayExpr), and [colorExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#colorExpr) properties, respectively.

- [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr)             

>     A data field used to assign instances of this resource kind to appointments. Add this field to appointment objects and set its value to resource instance `id`s.

Reviewer 4: ```suggestion
    A data field used to assign instances of this resource kind to appointments. Add this field to appointment objects and set the field value to a resource instance's `id` values.
```

> In this demo, the **resources** array contains three resource kinds: rooms, priorities, and owners. Their **fieldExpr** values are `roomId`, `priorityId`, and `ownerId`. Each appointment contains the same fields. Field values assign the appointments to different instances of these resource kinds.

Reviewer 4: ```suggestion
In this demo, the **resources** array contains three resource kinds: rooms, priorities, and owners. Their **fieldExpr** values are `roomId`, `priorityId`, and `ownerId`, respectively. Each appointment contains the same fields. Field values assign the appointments to different instances of these resource kinds.
```

> ### Color Appointments Based on Resource Kind

Reviewer 4: ```suggestion
### Color Appointments Based on a Resource Kind
```

To use the color scheme of a specific resource kind, enable the kind's [useColorAsDefault](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#useColorAsDefault) property. Otherwise, appointments use the color scheme of the last resource kind declared in the **resources** array.

This demo enables you to change the **useColorAsDefault** property at runtime. Click the radio buttons under the Scheduler to switch between different color schemes.

### Assign Multiple Instances of a Resource Kind

> Each resource kind object can contain the [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property. When it is **true**, users can assign multiple instances of this kind to a single appointment. In this demo, the Owner resource kind allows multiple instances.

Reviewer 4: ```suggestion
Each resource kind object can contain the [allowMultiple](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#allowMultiple) property. When this property is set to **true**, users can assign multiple instances of this kind to a single appointment. In this demo, the Owner resource kind allows multiple instances.
```
